jormungandr/src/network/mod.rs

```diff
@@ -309,7 +322,11 @@ async fn handle_network_input(
             } => {
                 state
                     .peers
-                    .pull_headers(node_address, from.into(), to)
+                    .pull_headers(
+                        node_address,
+                        try_ids_from_iter(from.as_slice()).unwrap(),
+                        BlockId::try_from_network(to).unwrap(),
```

Comment:
Is this `unwrap` ok? or should we skip is something goes wrong? I wasn't sure of it.

Reply:
Yes, conversion to the network types from internal types should always be correct.

Reply:
... and this is why the `IntoNetwork` trait should be used here instead.